contracts/oracle/MarketGovernanceOracle.sol

```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import {PCVOracle} from "./PCVOracle.sol";
+import {CoreRefV2} from "./../refs/CoreRefV2.sol";
+import {IVoltSystemOracle} from "./IVoltSystemOracle.sol";
+
+/// @notice contract that receives a fixed interest rate upon construction,
+/// and then linearly interpolates that rate over a 30.42 day period into the VOLT price
+/// after the oracle start time.
+/// interest can be compounded at any point in time if the Volt rate changes due to
+/// excessive illiquid composition of PCV.
+/// @author Elliot Friedman
+contract MarketGovernanceOracle is CoreRefV2 {
+    using SafeCast for *;
+
+    /// @notice event emitted when the Volt system oracle compounds
+    /// emits the end time of the period that completed and the new oracle price
+    event InterestCompounded(uint256 periodStart, uint256 newOraclePrice);
+
+    /// @notice event emitted when the Volt system oracle actual rate updates
+    event ActualRateUpdated(uint256 oldRate, uint256 newRate);
+
+    /// @notice event emitted when reference to the PCV Oracle updates
+    event PCVOracleUpdated(address oldOracle, address newOracle);
+
+    /// ---------- Mutable Variables ----------
+
+    /// @notice acts as an accumulator for interest earned in previous periods
+    /// returns the oracle price from the end of the last period
+    uint256 public oraclePrice;
+
+    /// @notice start time at which point interest will start accruing, and the
+    /// current ScalingPriceOracle price will be snapshotted and saved
+    uint256 public periodStartTime;
+
+    /// @notice current amount that oracle price is inflating by monthly as a percent, scaled by 1e18
+    /// set by governance, not actual rate Volt increases by
+    uint256 public baseChangeRate;
+
+    /// @notice actual amount that oracle price is inflating by monthly as a percent, scaled by 1e18
+    /// derived from base rate volt price interpolates by this amount.
+    uint256 public actualChangeRate;
+
+    /// @notice reference to the PCV oracle smart contract
+    address public pcvOracle;
+
+    /// ---------- Immutable Variable ----------
+
+    /// @notice the percentage at which the Volt rate will start going above the base rate
```

Comment:
should be  `the ratio of illiquid to liquid PCV amounts at which the Volt rate will start going above the base rate`